capa/features/extractors/dnfile/helpers.py

```diff
@@ -41,8 +41,72 @@ def seek(self, offset: int) -> int:
         return self.offset
 
 
-def calculate_dotnet_token_value(table: int, rid: int) -> int:
-    return ((table & 0xFF) << Token.TABLE_SHIFT) | (rid & Token.RID_MASK)
+class DnClass(object):
+    def __init__(self, token: int, namespace: str, classname: str):
+        self.token: int = token
+        self.namespace: str = namespace
+        self.classname: str = classname
+
+    def __hash__(self):
+        return hash((self.token,))
+
+    def __eq__(self, other):
+        return self.token == other.token
+
+    def __str__(self):
+        return DnClass.format_name(self.namespace, self.classname)
+
+    def __repr__(self):
+        return str(self)
+
+    @staticmethod
+    def format_name(namespace: str, classname: str):
+        name: str = classname
+        if namespace:
+            # like System.IO.File::OpenRead
+            name = f"{namespace}.{name}"
+        return name
+
+
+class DnMethod(DnClass):
+    def __init__(self, token: int, namespace: str, classname: str, methodname: str):
+        super(DnMethod, self).__init__(token, namespace, classname)
+        self.methodname: str = methodname
+
+    def __str__(self):
+        return DnMethod.format_name(self.namespace, self.classname, self.methodname)
+
+    @staticmethod
+    def format_name(namespace: str, classname: str, methodname: str):  # type: ignore
```

Comment:
mypy doesn't like overriding the static method with different argument types; fixing requires reorganizing the class hierarchy which I'd like to avoid as it over complicates the purpose here. if it becomes a issue in the future I'm happy to make changes as necessary.